src/data/GameContext.cpp

```diff
@@ -504,6 +504,39 @@ Achievement& GameContext::NewAchievement(AchievementSet::Type nType)
     return pAchievement;
 }
 
+bool GameContext::RemoveAchievement(unsigned int nAchievementId) noexcept
+{
+    for (auto pIter = m_vAchievements.begin(); pIter != m_vAchievements.end(); ++pIter)
```

Comment:
How about `for (auto pIter : m_vAchievements)`? I believe this should work as well and it is a little bit less boilerplate-y.

Reply:
It seems he needs the iterator for line 532.

Reply:
Using the shorthand loop eliminates the iterator variable, which is needed for the erase call  (pIter would be a reference to an element).

Reply:
I see, my bad.